rotkehlchen/chain/ethereum/structures.py

```diff
@@ -357,6 +379,105 @@ def serialize(self) -> Dict[str, Any]:
             'log_index': self.log_index,
         }
 
+    def serialize_for_db(self, address: ChecksumEthAddress) -> YEARN_EVENT_DB_TUPLE:
+        pnl_amount = None
+        pnl_usd_value = None
+        if self.realized_pnl:
+            pnl_amount = str(self.realized_pnl.amount)
+            pnl_usd_value = str(self.realized_pnl.usd_value)
+        vault_version = self.version
+        return (
+            address,
+            self.event_type,
+            self.from_asset.identifier,
+            str(self.from_value.amount),
+            str(self.from_value.usd_value),
+            self.to_asset.identifier,
+            str(self.to_value.amount),
+            str(self.to_value.usd_value),
+            pnl_amount,
+            pnl_usd_value,
+            str(self.block_number),
+            str(self.timestamp),
+            self.tx_hash,
+            self.log_index,
+            vault_version,
+        )
+
+    @classmethod
+    def deserialize_from_db(cls, result: YEARN_EVENT_DB_TUPLE) -> 'YearnVaultEvent':
+        """
+        Turns a tuple read from the DB into an appropriate YearnVaultEvent
+        May raise a DeserializationError if there is an issue with information in
+        the database
+        """
+        location = 'deserialize yearn vault event from db'
+        realized_pnl = None
+        if result[8] is not None and result[9] is not None:
+            pnl_amount = deserialize_optional_fval(
+                value=result[8],
+                name='pnl_amount',
+                location=location,
+            )
+            pnl_usd_value = deserialize_optional_fval(
+                value=result[9],
+                name='pnl_usd_value',
+                location=location,
+            )
+            realized_pnl = Balance(amount=pnl_amount, usd_value=pnl_usd_value)
+
+        from_value_amount = deserialize_optional_fval(
+            value=result[3],
+            name='from_value_amount',
+            location=location,
+        )
+        from_value_usd_value = deserialize_optional_fval(
+            result[4],
+            name='from_value_usd_value',
+            location=location,
+        )
+        to_value_amount = deserialize_optional_fval(
+            value=result[6],
+            name='to_value_amount',
+            location=location,
+        )
+        to_value_usd_value = deserialize_optional_fval(
+            value=result[7],
+            name='to_value_usd_value',
+            location=location,
+        )
+        try:
+            block_number = int(result[10])
+        except ValueError as e:
+            raise DeserializationError(
+                f'Failed to deserialize block number {result[10]} in yearn vault event: {str(e)}',
+            ) from e
+        try:
+            from_asset = Asset(result[2])
+        except UnknownAsset as e:
+            raise DeserializationError(
+                f'Failed to deserialize from_asset {result[2]} in yearn vault event',
+            ) from e
```

Comment:
Don't do that. Just let UnknownAsset also propagate upwards and handle it where this function is called.